src/main/java/com/netflix/imflibrary/writerTools/IMPAssembler.java

```diff
@@ -0,0 +1,387 @@
+        private List<ErrorLogger.ErrorObject> errors;
+
+        /**
+         * Constructor for an assembled IMP result
+         * @param cpl - the CPL file
+         * @param pkl - the PKL file
+         * @param assetMap - the AssetMap file
+         *
+         * @param errors - a list of errors encountered when creating the IMP
+         */
+        public AssembledIMPResult(File cpl, File pkl, File assetMap, List<File> mediaFiles, List<ErrorLogger.ErrorObject> errors) {
+            this.cpl = cpl;
+            this.pkl = pkl;
+            this.assetMap = assetMap;
+            this.mediaFiles = mediaFiles;
+            this.errors = errors;
+        }
+    }
+
+
+    public static class SimpleTimeline {
+        public SimpleTimeline() {
+            this.tracks = new ArrayList<>();
+        }
+
+        /**
+         * Constructor for a simple timeline
+         * @param tracks - a list of tracks to use in the timeline
+         * @param editRate - the edit rate, must match the video frame rate
+         */
+        public SimpleTimeline(List<Track> tracks, Composition.EditRate editRate) {
+            this.tracks = tracks;
+            this.editRate = editRate;
+        }
+
+        public void setTracks(List<Track> tracks) {
+            this.tracks = tracks;
+        }
+
+        public void setEditRate(Composition.EditRate editRate) {
+            this.editRate = editRate;
+        }
+
+        public List<Track> getTracks() {
+            return tracks;
+        }
+
+        public Composition.EditRate getEditRate() {
+            return editRate;
+        }
+
+        private List<Track> tracks;
+        public Composition.EditRate editRate;
+    }
+
+    public static class Track {
+        public Track() {
+            this.mediaFiles = new ArrayList<>();
+        }
+
+        /**
+         * Constructor for a track to be used to construct a simple timeline
+         * @param mediaFiles - a list of media files to use in the track, can contain edits
+         * @param sequenceTypeEnum - describes whether the track is a video, audio, etc..
+         */
+        public Track(List<MediaFile> mediaFiles, Composition.SequenceTypeEnum sequenceTypeEnum) {
+            this.mediaFiles = mediaFiles;
+            this.sequenceTypeEnum = sequenceTypeEnum;
+        }
+
+        public Composition.SequenceTypeEnum getSequenceTypeEnum() {
+            return sequenceTypeEnum;
+        }
+
+        public void setSequenceTypeEnum(Composition.SequenceTypeEnum sequenceTypeEnum) {
+            this.sequenceTypeEnum = sequenceTypeEnum;
+        }
+
+        public List<MediaFile> getMediaFiles() {
+            return mediaFiles;
+        }
+
+        public void setMediaFiles(List<MediaFile> mediaFiles) {
+            this.mediaFiles = mediaFiles;
+        }
+
+        public Composition.SequenceTypeEnum sequenceTypeEnum;
+        public List<MediaFile> mediaFiles;
+    }
+
+    public static class MediaFile {
```

Comment:
I would suggest that we rename this to something other than `'MediaFile'`. The term 'file' may be misleading, because what this class _actually_ holds is more than just a file. It also includes timing and edit information used to define an entry in the track.  Maybe the name 'TrackEntry' would be more appropriate?

-----

In IMF terminology, within a CPL this is called a `Resource`. A resource identifies the specific MXF file ('TrackFile'), but also how it is being used within the track for this particular instance. Multiple _Resources_ can point to the same MXF file.  Example:
- Resource1:  `mainVideo_01.mxf`,  frames 0-500  _// Uses the first 500 frames of one file_
- Resource2: `videoInsert.mxf`, frames 0-48  _// Adds the first 48 frames from a different file_
- Resource3: `mainVideo_01.mxf`, frames 548-3000  _// Back to the first file again, skipping the next 48 frames_

In the above example, `mainVideo_01.mxf` is used twice. The first and third entries in the track are different `Resources`, but they reference the same 'TrackFile'.

-----

If I understand correctly, your definition of `MediaFile` is trying to do something similar to the CPL `Resource`, right? It is possible for the same MXF file to be present in _multiple_ `MediaFiles` per Track, correct? If that is the case, it might make sense to avoid using the word 'file' in the class name, just for clarity.

Reply:
Thanks @davidt-netflix I'll rename `MediaFile` to `TrackFile`.. It _is_ possible to use the same file multiple times in the track - the unit test does this. I'll update the class name.

Reply:
@davidt-netflix , I overwrote the previous commit and addressed the issues you mentioned. Please let me know what you think!